### PR TITLE
refactor(docs): make discord-js docs commands dynamically

### PR DESCRIPTION
**Pull request by JPBM135**
## Why?
The parser used had a big problem, [`stable,json`](https://github.com/discordjs/docs/blob/main/discord.js/stable.json) isn't updated (v13) and [`main.json`](https://github.com/discordjs/docs/blob/main/discord.js/main.json) is too updated.

This PR enables the bot to lock the versions using version-specific JSONs and enables the maintainers to dynamically update versions during minor updates using the `/npmreload` command

- `discord-js` => `discord-js-v13`
- **New:** `discord-js-v14`
- `discord-js-dev`

<details><summary>Demonstrations</summary>

image

image
</details>

### CODE REVIEW

**Review by almostSouji — CHANGES_REQUESTED** [review]

File: src/deployFunctions/deployGlobal.ts [hunk 1]
```diff
@@ -4,7 +4,17 @@ import { GuideCommand } from '../interactions/guide';
...
-void deploy([DiscordDocsCommand, DocsCommand, GuideCommand, InviteCommand, MdnCommand, NodeCommand, TagCommand]);
+void deploy([
+	DiscordDocsCommand,
+	DocsCommand,
+	GuideCommand,
+	InviteCommand,
+	MdnCommand,
+	NodeCommand,
+	TagCommand,
+	NpmReloadCommand,
```
[the remark this answers is not in the document]
Comment (JPBM135) [reply src/deployFunctions/deployGlobal.ts]: Reverted in 843641d

File: src/functions/npm.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	customSources.set('v13-lts', formatVersionToRaw(json['dist-tags']['v13-lts']));
+	customSources.set('latest', formatVersionToRaw(json['dist-tags'].latest));
+}
+
+export async function reloadNpmVersions(res: Response, customSources: Map<CustomSourcesString, string>) {
+	const prev = {
+		v13: customSources.get('v13-lts')!,
+		latest: customSources.get('latest')!,
+	};
+	customSources.clear();
```
Comment (JPBM135) [inline src/functions/npm.ts:32]: Remove this, is unnecessary!
```suggestion
```

File: src/functions/npm.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+			latest: customSources.get('latest')!,
+		};
+
+		prepareResponse(
+			res,
+			`${PREFIX_SUCCESS} **Npm versions updated!**\nPrevious versions: \`\`\`json\n${JSON.stringify(
+				prev,
+				null,
+				4,
+			)}\`\`\`\nNew versions: \`\`\`json\n${JSON.stringify(newVersions, null, 4)}\`\`\``,
```
Comment (JPBM135) [inline src/functions/npm.ts:47]: Improve this message formating

File: src/index.ts [hunk 4]
```diff
@@ -68,7 +70,9 @@ async function verify(req: Request, res: Response, next: NextHandler) {
 
 const tagCache = new Collection<string, Tag>();
 const mdnIndexCache: MDNIndexEntry[] = [];
+const customSources = new Map<CustomSourcesString, string>();
 void loadTags(tagCache);
+void loadLatestNpmVersion(customSources);
 logger.info(`Tag cache loaded with ${tagCache.size} entries.`);
```
Comment (JPBM135) [inline src/index.ts:76]:
```suggestion
void loadTags(tagCache);
logger.info(`Tag cache loaded with ${tagCache.size} entries.`);
await loadLatestNpmVersion(customSources);
logger.info({
    msg: 'Loaded latest npm versions for discord.js',
    ['v13-lts']: customSources.get('v13-lts'),
    latest: customSources.get('latest'),
})
```

File: src/functions/npm.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import { request } from 'undici';
+import { CustomSourcesString } from '../types/discordjs-docs-parser';
+import { PartialNpmAPIResponse } from '../types/npm';
+import { logger, PREFIX_SUCCESS, prepareErrorResponse, prepareResponse } from '../util';
+
+function formatVersionToRaw(version: string): string {
+	return `https://raw.githubusercontent.com/discordjs/docs/main/discord.js/${version}.json`;
+}
+
+export async function loadLatestNpmVersion(customSources: Map<CustomSourcesString, string>): Promise<void> {
```
Comment (JPBM135) [inline src/functions/npm.ts:11]: Add an Util function to check if the files are valid/exist

**Comment by almostSouji** [comment]
needs a rebase

**Comment by JPBM135** [comment]
> needs a rebase

Okie

File: src/interactions/docs.ts [hunk 6]
```diff
@@ -10,8 +10,28 @@ export const DocsCommand = {
...
+					name: 'target',
+					description: TARGET_DESCRIPTION,
+					required: false,
+				},
+			],
+		},
+		{
+			type: ApplicationCommandOptionType.Subcommand,
+			name: 'discord-js-v14',
+			description: `${SUBCOMMAND_DESCRIPTION_PREFIX} discord.js@latest`,
```
Comment (jaw0r3k) [inline src/interactions/docs.ts:34]: you should so a rebase 
and add `hide` options

File: src/interactions/docs.ts [hunk 7]
```diff
@@ -11,8 +11,28 @@ export const DocsCommand = {
...
+					description: QUERY_DESCRIPTION,
+					required: true,
+					autocomplete: true,
+				},
+				{
+					type: ApplicationCommandOptionType.User,
+					name: 'target',
+					description: TARGET_DESCRIPTION,
+					required: false,
+				},
```
Comment (jaw0r3k) [inline src/interactions/docs.ts:29]:
```suggestion
				},
				{
					type: ApplicationCommandOptionType.Boolean,
					name: 'hide',
					description: EPHEMERAL_DESCRIPTION,
					required: false,
				},
			],
```

Comment (JPBM135) [reply src/interactions/docs.ts:29]: Will do!

Comment (JPBM135) [reply src/interactions/docs.ts:29]: Done in e60978a

**Review by almostSouji — APPROVED** [review]

**Pull request merged**